presto-main/src/main/java/com/facebook/presto/operator/scalar/VarbinaryFunctions.java

```diff
@@ -251,4 +252,75 @@ public static long crc32(@SqlType(StandardTypes.VARBINARY) Slice slice)
         crc32.update(slice.toByteBuffer());
         return crc32.getValue();
     }
+
+    @Description("suffix starting at given index")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice substr(@SqlType(StandardTypes.VARBINARY) Slice slice, @SqlType(StandardTypes.BIGINT) long start)
+    {
+        if (start == 0 || slice.length() == 0) {
+            return Slices.EMPTY_SLICE;
+        }
+
+        int startByte = Ints.saturatedCast(start);
+
+        if (startByte > 0) {
+            int indexStart = startByte - 1; // index starts with 1.
+            if (indexStart >= slice.length()) {
+                return Slices.EMPTY_SLICE;
+            }
+            return slice.slice(indexStart, slice.length() - indexStart);
+        }
+
+        // negative start is relative to end of string
+        startByte += slice.length();
+
+        // before beginning of string
+        if (startByte < 0) {
+            return Slices.EMPTY_SLICE;
+        }
+
+        return slice.slice(startByte, slice.length() - startByte);
+    }
+
+    @Description("substring of given length starting at an index")
```

Comment:
How about calling it 'sub binary string'?

Reply:
Let's leave this one alone